tardis/utilities/executors/sshexecutor.py

```diff
@@ -3,43 +3,107 @@
 from ...interfaces.executor import Executor
 from ..attributedict import AttributeDict
 
+try:
+    from contextlib import asynccontextmanager
+except ImportError:
+    from aiotools import async_ctx_manager as asynccontextmanager
+
 import asyncssh
+import asyncio
+
+
+class SSHConnectionPool(object):
+    _connection_pool = dict()
```

Comment:
I'm not sure that this should be "global" instead of per-instance. It seems that there is one ``SSHConnectionPool`` instance per remote host+parameter, so the instance could just hold its own pool.

Most importantly, we may end up having different parameters per host. Having a separate "pool" per instance makes that trivial; a global pool means we get conflicts and the first host+parameter set wins.